### PR TITLE
feat: docker build workflow

### PR DESCRIPTION
**Pull request by sammcj**
- Build container image
- Publish to Github Container Registry
- Fix deprecated Dockerfile syntax
- Minor linting fixes

Tested here - https://github.com/sammcj/explainshell/actions/runs/3708331408

@idank

### CODE REVIEW

**Comment by sammcj** [comment]
@idank

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,18 @@
 FROM python:2.7
 
-MAINTAINER Simon Toivo Telhaug <simon.toivo@gmail.com>
+LABEL AUTHOR="Simon Toivo Telhaug <simon.toivo@gmail.com>"
```
Comment (idank) [inline Dockerfile:3]: let's remove these two lines

**Comment by idank** [comment]
Looks great overall, left one minor comment. Thanks for contributing!

File: Dockerfile (hunk 1 above)
Comment (sammcj) [reply Dockerfile]: Done

**Comment by idank** [comment]
Can you squash the commits please?

**Comment by sammcj** [comment]
Usually GitHub has the squash and merge when merging PRs?

If it's not showing up for some reason I'll do it in a few days as I'm away from my computer for the holidays.

**Comment by idank** [comment]
oh you're right, i haven't seen that before, merging!

happy holidays and thanks again

**Comment by idank** [comment]
It appears there's some breakage in the tests workflow, I couldn't find the cause at a quick glance, any idea what changed?

https://github.com/idank/explainshell/actions/runs/3771229887/jobs/6411513291#step:8:170

**Pull request merged**